docs/source/conf.py

```diff
@@ -441,15 +446,552 @@ def check_timeout_and_api_kwargs_presence(obj: object) -> int:
     )
 
 
+class AdmonitionInserter:
+    """Class for inserting admonitions into docs of Telegram classes."""
+
+    # We cannot move this code to a separate module within docs because there is no way
+    # to import it.  This class is simply a way of encapsulating the code and providing a better
+    # overview of conf.py.
+
+    ADMONITION_TYPES = ("use_in", "available_in", "returned_in")
+
+    # In some cases it is useful that some admonitions are displayed not only for base classes
+    # but also for their subclasses.
+    CLASSES_WHOSE_SUBCLASSES_NEED_ADMONITIONS = (
+        telegram.BotCommandScope,
+        telegram.InlineQueryResult,
+        telegram.InputMedia,
+        telegram.MenuButton,
+        telegram.request.BaseRequest,
+    )
```

Comment:
Maybe I missed part of the discussion … Can't this be automated or were there arguments against doing that? Keeping this list updated is one more thing to remember …

Reply:
Clarification: I'd like to have the "link all subclasses wherever the base class is mentioned" logic applied to all classes. Unless there are specific reasons not to do that.

Reply:
I can see that for "Use in" admonitions. Maybe for "Available in", not sure. But for "Return in"? See [this comment](https://github.com/python-telegram-bot/python-telegram-bot/pull/3501#issuecomment-1398898882).

Reply:
>See [this comment](https://github.com/python-telegram-bot/python-telegram-bot/pull/3501#issuecomment-1398898882).

check dev chat

Reply:
Done